koji_containerbuild/plugins/builder_containerbuild.py

```diff
@@ -263,10 +243,18 @@ def get_additional_tags(self):
     def get_missing_label_ids(self):
         data = self.get_data_labels()
         missing_labels = []
-        for label_id in LABELS:
+        # check required labels, have to be defined explicitly and not via env
+        for label_id in LABELS_EXPLICIT_REQUIRED:
             assert label_id in data
             if not data[label_id]:
                 missing_labels.append(label_id)
+
+        # check for all labels, unless required or default values provided,
+        # they should be at least defined (even via env)
+        for label_id in LABELS_ENV_REQUIRED:
+            assert label_id in data
+            if data[label_id] is None:
```

Comment:
I think this is a bit subtle, and takes a bit of staring at to work out why it works.

* we set `data[label_id]` to `None` earlier on, in get_data_labels
* we do *not* have the parent environment available in order to correctly substitute environment variables in label values, so e.g. `LABEL foo="$BAR"` will end up with data['foo'] being the empty string
* it's the missing_labels variable, and not the assert, which is the actual check.

Would it be more readable if we made the following changes?:
* don't set explicit `None` values in get_data_labels, just leave them out of the dictionary if they aren't in the Dockerfile
* remove the two asserts here, because they are not the checks we care about
* replace the checks with:
  * `if not data.get(label_id):` (for `LABELS_EXPLICIT_REQUIRED`) and
  * `if label_id in data:` (for `LABELS_ENV_REQUIRED`)

Reply:
chaned